### PR TITLE
feat(nexus): allow shared replica to be added as local child

### PR DESCRIPTION
**Pull request by jonathan-teh**
Fixes CAS-748.

### CODE REVIEW

**Comment by jonathan-teh** [comment]
See also openebs/spdk@b0768fd.

**Comment by jonathan-teh** [comment]
bors try

**Review by cjones1024 — APPROVED** [review]

File: mayastor/src/target/iscsi.rs [hunk 1]
```diff
@@ -233,6 +233,9 @@ fn share_as_iscsi_target(
 /// Export given bdev over iscsi. That involves creating iscsi target and
 /// adding the bdev as LUN to it.
 pub fn share(bdev_name: &str, bdev: &Bdev, side: Side) -> Result<String> {
+    if bdev.is_claimed() {
+        return Err(Error::CreateTarget {});
```
Comment (tiagolobocastro) [inline mayastor/src/target/iscsi.rs:237]: Should this have a message like the nvmf?

**Review by tiagolobocastro — APPROVED** [review]
cool! :+1:

**Review by jkryl — APPROVED** [review]

File: mayastor/src/target/iscsi.rs (hunk 1 above)
Comment (jonathan-teh) [reply mayastor/src/target/iscsi.rs]: Done, for consistency with nvmf.

**Comment by jonathan-teh** [comment]
bors merge

**Pull request merged**